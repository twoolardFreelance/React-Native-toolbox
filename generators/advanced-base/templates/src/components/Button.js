// @flow
import React, { Component } from 'react';
import { Text, StyleSheet } from 'react-native';

import Touchable from './Touchable';

import theme from 'theme';

export type Props = {
  text?: string,
  onPress: () => void,
  children?: React.Element<*>,
  style?: any,
  textStyle?: any,
};

type DefaultProps = {
  onPress: () => void,
};

class Button extends Component<DefaultProps, Props, void> {
  static defaultProps: DefaultProps = {
    onPress: () => {},
  };

  render() {
    const { style, text, textStyle, children, ...rest } = this.props;
    const content = text
      ? <Text style={[styles.text, textStyle]}>
          {text.toUpperCase()}
        </Text>
      : children;

    return (
      <Touchable style={[styles.button, style]} {...rest}>
        {content}
      </Touchable>
    );
  }
}

const styles = StyleSheet.create({
  button: {
    alignSelf: 'stretch',
    justifyContent: 'center',
    height: 40,
    backgroundColor: theme.colors.primary,
    paddingHorizontal: 8,
    borderRadius: 5,
    marginVertical: 8,
  },
  text: {
    ...theme.fonts.button,
    color: theme.colors.overPrimary,
    textAlign: 'center',
  },
});

export default Button;
