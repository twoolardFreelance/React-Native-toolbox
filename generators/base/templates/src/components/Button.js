import React, { PropTypes } from 'react';
import { TouchableOpacity, Text, StyleSheet, View } from 'react-native';

import appStyle from '<%= appName %>/src/appStyle';

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    minHeight: appStyle.dimensions.touchableHeight,
    marginVertical: appStyle.margins.inner,
  },
  button: {
    alignSelf: 'stretch',
    justifyContent: 'center',
    height: appStyle.dimensions.visibleButtonHeight,
    backgroundColor: appStyle.colors.primary,
    paddingHorizontal: appStyle.margins.inner,
  },
  text: {
    textAlign: 'center',
    color: appStyle.colors.lightText,
    fontSize: appStyle.font.fontSize.default,
  },
});

const Button = props => (
  <TouchableOpacity onPress={props.onPress} style={styles.container}>
    <View style={styles.button}>
      <Text style={[styles.text]}>{props.children.toUpperCase()}</Text>
    </View>
  </TouchableOpacity>
);

Button.propTypes = {
  children: PropTypes.string,
  onPress: PropTypes.func,
  buttonType: PropTypes.string,
};

export default Button;
