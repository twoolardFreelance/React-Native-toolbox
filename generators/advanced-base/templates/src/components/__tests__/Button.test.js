import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../Button';

describe('Button', () => {
  it('should render the button', () => {
    const tree = renderer.create(<Button text="Hello" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
