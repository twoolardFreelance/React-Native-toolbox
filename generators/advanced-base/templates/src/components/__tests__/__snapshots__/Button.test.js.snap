// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "opacity": 1,
        },
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#3f51b5",
            "borderRadius": 5,
            "height": 40,
            "justifyContent": "center",
            "marginVertical": 8,
            "paddingHorizontal": 8,
          },
          undefined,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "700",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      HELLO
    </Text>
  </View>
</View>
`;
